lib/rust/parser/debug/tests/parse.rs

```diff
@@ -1189,6 +1189,35 @@ fn pattern_match_auto_scope() {
     test(&code.join("\n"), expected);
 }
 
+// === Private (project-private) keyword ===
+#[test]
+fn private_keyword() {
+    test("private", block![(Private)]);
+}
+
+#[test]
+#[ignore]
+fn private_is_first_statement() {
+    // Comments and empty lines are allowed before `private`.
+    #[rustfmt::skip]
+    let lines = vec![
+        "# Some comment",
+        "# Other comment",
```

Comment:
We can just check for `Documented` nodes and treat them as non-contentful. They are bodiless during macro resolution because that type of multi-line construct is assembled after macros are run.